src/python/alog/protocols.py

```diff
@@ -0,0 +1,576 @@
+    handlers: List["Handler"]  # undocumented
+    disabled: bool  # undocumented
+    root: ClassVar["RootLogger"]  # undocumented
+    manager: "Manager"  # undocumented
+
+    def __init__(self, name: str, level: "_Level" = 0) -> None:
+        """
+        Initialize the logger with a name and an optional level.
+        """
+
+    def setLevel(self, level: "_Level") -> None:  # pylint: disable=invalid-name
+        """
+        Set the logging level of this logger.  level must be an int or a str.
+        """
+
+    def isEnabledFor(self, level: int) -> bool:  # type: ignore pylint: disable=invalid-name
+        """
+        Is this logger enabled for level 'level'?
+        """
+
+    def getEffectiveLevel(self) -> int:  # type: ignore pylint: disable=invalid-name
+        """
+        Get the effective level for this logger.
+
+        Loop through this logger and its parents in the logger hierarchy,
+        looking for a non-zero logging level. Return the first one found.
+        """
+
+    def getChild(self, suffix: str) -> "Self":  # type: ignore pylint: disable=invalid-name
+        """
+        Get a logger which is a descendant to this one.
+
+        This is a convenience method, such that
+
+        logging.getLogger('abc').getChild('def.ghi')
+
+        is the same as
+
+        logging.getLogger('abc.def.ghi')
+
+        It's useful, for example, when the parent logger is named using
+        __name__ rather than a literal string.
+        """
+        # see python/typing#980
+
+    if sys.version_info >= (3, 12):
+
+        def getChildren(self) -> Set["Logger"]:  # pylint: disable=invalid-name
+            ...
+
+    # Aligned with the typeshed type definition:
+    # `*kwargs` replaced with: `exc_info`, `stack_info`, `stacklevel`, `extra`
+    def debug(
+        self,
+        msg: object,
+        *args: object,
+        exc_info: "_ExcInfoType" = None,
+        stack_info: bool = False,
+        stacklevel: int = 1,
+        extra: Optional[Mapping[str, object]] = None,
+    ) -> None:
+        """
+        Log 'msg % args' with severity 'DEBUG'.
+
+        To pass exception information, use the keyword argument exc_info with
+        a true value, e.g.
+
+        logger.debug("Houston, we have a %s", "thorny problem", exc_info=1)
+        """
+
+    # Aligned with the typeshed type definition:
+    # `*kwargs` replaced with: `exc_info`, `stack_info`, `stacklevel`, `extra`
+    def info(
+        self,
+        msg: object,
+        *args: object,
+        exc_info: "_ExcInfoType" = None,
+        stack_info: bool = False,
+        stacklevel: int = 1,
+        extra: Optional[Mapping[str, object]] = None,
+    ) -> None:
+        """
+        Log 'msg % args' with severity 'INFO'.
+
+        To pass exception information, use the keyword argument exc_info with
+        a true value, e.g.
+
+        logger.info("Houston, we have a %s", "interesting problem", exc_info=1)
+        """
+
+    if sys.version_info >= (3, 12):
```

Comment:
The only difference I see here is `interesting` vs `notable`. Is there something else I'm seeing behind this version switch?

Reply:
Ah, I see that this is designed for exact docstring compatibility!